### PR TITLE
[JENKINS-30040] Logging exception running triggering nonblocking builds

### PR DESCRIPTION
**Pull request by svanoort**
Builds on top of fix in [PR #90](https://github.com/jenkinsci/parameterized-trigger-plugin/pull/90) but includes tests.  I've manually verified this works correctly from the UI as well. 

@tonyfinn @reviewbybees

### CODE REVIEW

File: src/main/java/hudson/plugins/parameterizedtrigger/TriggerBuilder.java [hunk 1]
```diff
@@ -157,10 +158,12 @@ public boolean perform(AbstractBuild<?, ?> build, Launcher launcher,
         return buildStepResult;
     }
 
-    private String getProjectListAsString(List<Job> projectList){
+    // Public but restricted so we can add tests without completely changing the tests package
+    @Restricted(value=org.kohsuke.accmod.restrictions.NoExternalUse.class)
```
Comment (oleg-nenashev) [inline src/main/java/hudson/plugins/parameterizedtrigger/TriggerBuilder.java]: NIT: would be better to just import the class, value= is also not required

**Comment by oleg-nenashev** [comment]
:bee:

**Comment by tfennelly** [comment]
:bee:

**Comment by svanoort** [comment]
@reviewbybees done.

**Comment by svanoort** [comment]
@ikedam Any thoughts on this, as maintainer?

**Comment by ikedam** [comment]
@svanoort 
I no longer maintain this plugin.
Please contact the current maintainer.

**Comment by svanoort** [comment]
Thanks!  

@olivergondza Would you have any issues if this were merged / released?  It's a fairly trivial fix for an irritating bug.

**Comment by olivergondza** [comment]
Releasing 2.29 right now.

**Pull request merged**